lib/ansible/plugins/inventory/foreman.py

```diff
@@ -121,134 +126,124 @@ def _get_session(self):
             self.session.verify = self.get_option('validate_certs')
         return self.session
 
-    def _get_json(self, url, ignore_errors=None):
+    def _get_all_hosts(self):
 
-        if not self.use_cache or url not in self._cache.get(self.cache_key, {}):
+        if self.cache:
```

Comment:
Use `self._cache` instead of `self.cache`. I'm not sure what `self.cache.get_inventory()` is, but you should use `self._cache[cache_key]` to access the inventory for this particular plugin. There's documentation you can follow here: https://docs.ansible.com/ansible/latest/dev_guide/developing_inventory.html#inventory-cache

Reply:
It seems like this part is a leftover of my previous implementation where I extended the cache layer.